### PR TITLE
Update header view when the topic is changed

### LINKED ISSUE #20167: Reader: the following status on a Reader site is not changed when tapped
### Expected behavior
The following status button is changed immediately after being tapped.

### Actual behavior
The following status button is changed after the follow or unfollow site API call is returned

### Steps to reproduce the behavior
1. Go to the Reader tab
2. Go to a site from there
3. Tap the following status ("Following" or "Follow") button

##### Tested on [device], iOS [version], WPiOS [version]
8ee084a2df913ff5fb6a02e4ec859eed5cfb9e30

### PR DESCRIPTION
**Pull request by crazytonyli**
Fix #20167, which was introduced in #20150.

## Regression Notes
1. Potential unintended areas of impact
None.

5. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

6. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 1]
```diff
@@ -355,6 +357,14 @@ import Combine
...
             displayLoadingStream()
         }
+
+        if let readerTopic {
+            readerTopic.objectWillChange
+                .sink { [weak self] _ in
+                    self?.updateStreamHeaderIfNeeded()
+                }
+                .store(in: &cancellables)
+        }
```
Comment (mokagio) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift:367]: Did you consider optional chaining?

```suggestion
        readerTopic?
            .objectWillChange
            .sink { [weak self] _ in
                self?.updateStreamHeaderIfNeeded()
            }
            .store(in: &cancellables)
```

I tried it locally and it worked. 🤷‍♂️

**Review by mokagio — APPROVED** [review]
Lovely to see some Combine for a change ✨

**Comment by mokagio** [comment]
Taking over merging this so it lands on `trunk` before the 21.8 code freeze, as that's the milestone given to this PR.

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift (hunk 1 above)
Comment (crazytonyli) [reply WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift:367]: I did not 😓 . Neat idea though, I've created a follow up PR https://github.com/wordpress-mobile/WordPress-iOS/pull/20171

Comment (mokagio) [reply WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift:367]: Thank you for considering my suggestion.

**Pull request merged**